File: sumEqualKHashing.py
n = int(input("enter the length of array "))
*arr, = map(int, input().split())
sum = int(input("enter the sum value "))
c = 0
d={}
for val in arr:
    d[val] = sum - val

for val in d:
    try:
        if val == d[d[val]]:
            c += 1
    except:
        d[val] = 'not included'
print(c//2)

File: sumEqualKHashing.c
#include<stdio.h>
#include<malloc.h>
int main()
{
  int b[10000000] = {0};
  int n, temp, sum, c = 0, i;
  printf("enter the length of array\n");
  scanf("%d", &n);
  int *a = (int*)malloc(n * sizeof(int));
  for(i = 0; i < n; i++)
  {
    scanf("%d", &a[i]);
  }
  printf("enter the sum\n");
  scanf("%d", &sum);
  for(i = 0; i < n; i++)
  {
    temp = sum - a[i];
    if(b[temp] == 1)
    {
      c++;
    }
    b[a[i]] = 1;
  }
  printf("the no of pairs are %d", c);
	return 0;
}
